components/search/src/types.rs

```diff
@@ -127,9 +128,13 @@ pub struct SearchEngineUrl {
     /// is included in the base.
     pub search_term_param_name: Option<String>,
 
-    /// The display name of the URL, if any. This is useful if the URL
-    /// corresponds to a brand name distinct from the engine's brand name.
-    pub display_name: Option<String>,
+    /// A map from locale codes to display names of the URL. This is useful if
+    /// the URL corresponds to a brand name distinct from the engine's brand
+    /// name. Since brand names can be localized, this is a map rather than a
+    /// URL. The client will fall back to the special locale code "default" when
+    /// its locale is not present in the map.
+    #[uniffi(default = None)]
+    pub display_name_map: Option<HashMap<String, String>>,
```

Comment:
Thought: Since we already do processing based on the locale within the component, I'm wondering if we could should simplify this, and simply return the `display_name`. This would simplify the application implementation.

There's `user_environment.locale` which could be used and passed into `from_configuration_details`, however... this would complicate the `From<JSONEngineUrls>` functions...

Reply:
Yeah, that makes sense. I'll give it a shot.